hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeStateManager.java

```diff
@@ -708,6 +707,21 @@ private void updateNodeState(DatanodeInfo node, Predicate<Long> condition,
     }
   }
 
+  /**
+   *
+   * @param datanodeDetails
+   */
+  public void stopNode(DatanodeDetails datanodeDetails)
+      throws NodeNotFoundException {
+    UUID uuid = datanodeDetails.getUuid();
+    NodeState state = nodeStateMap.getNodeState(uuid);
+    DatanodeInfo node = nodeStateMap.getNodeInfo(uuid);
+    nodeStateMap.updateNodeState(datanodeDetails.getUuid(), state,
+        NodeState.DEAD);
+    node.updateLastHeartbeatTime(0L);
```

Comment:
Can you please explain why you set LastHeartbeatTime to 0?

Reply:
According to LastHeartbeatTime, NodeStateManager#checkNodesHealth detemine the status of node periodically.
If LastHeartbeatTime is not set to 0, NodeStateManager#checkNodesHealth could update the healty of node after we set the status of the node to dead.

Reply:
Yeah, i got it, i agree that it can works, but the downside, we lost the real lastHeartBeatTime.

Reply:
Yeah, this problem really exists.
i will create a new jira to solve this prolem.

Reply:
Ok, so would you like to create a ticket an paste it as a TODO comment here.